haas/client/base.py

```diff
@@ -0,0 +1,41 @@
+""" This module implements the HaaS client library. """
+
+import requests
+import os
+import json
+from urlparse import urljoin
+#from requests.exceptions import ConnectionError, MissingSchema
+
+from client_errors import *
+
+
+class ClientBase(object):
+    """ Main class which contains all the methods to
+    -- ensure input complies to API requisites
+    -- generates correct format for server API on behalf of the client
+    -- parses output from received from the server.
+    In case of errors recieved from server, it will generate appropriate
+    appropriate message.
+    """
+
+
+    def __init__(self, endpoint=None, auth=None):
```

Comment:
The default is there so as to get the setup as easy as possible to start with. 
Default being no authentication required.